core/src/main/java/hudson/PluginManager.java

```diff
@@ -219,6 +230,14 @@ public Api getApi() {
     }
 
     /**
+     * If non-null, the base directory for all exploded .hpi/.jpi plugins.
+     * @return the base directory for all exploded .hpi/.jpi plugins or {@code null} to leave this up to the strategy.
+     */
+    public File getWorkDir() {
```

Comment:
Well I considered `@CheckForNull`... but that would open the whole can of worms over **which** `@CheckForNull` the one that is in a package that is not incongruent with the Java platform policy on reserved package names, or the one claiming to be from a JSR that has never published anything - and thus is illegally squatting in the `javax` reserved namespace...

Also returning `rootDir` would change the behaviour of `ClassicPluginStrategy` which actually extracts into the same directory as the .jpi file is located and that need not necessarily be `rootDir` from the contract of a `PluginStrategy`